### DISCUSSION TITLE
"Quality" of 'grid-import' value in mein-senec.de Portal

### DISCUSSION THREAD

**Discussion by marq24** [marq24/ha-senec-v3#37] [General]
Hi,

just wanted to share with you my latest experience with the data available in mein-senec.de. Today I have switched my power provider and so I have recorded at 0:00 o'clock the electricity-meter value. The last time I was reading/recording the meter (manually) was on 1'st of January.

The value at the 1/Jan/23 was `1765 kWh` - the value at the end of 15/Oct/23 the `3654 kWh` - so the imported grid power is `1889 kWh`. My expectation was/is, when I use the [Data Details Page @ mein-senec.de](https://mein-senec.de/endkunde/#/0/status/gridimport/range/2023-10-15-288) for the the same period, that I should get similar numbers - or better my expectation is to get _identical_ numbers. But this is not the case! - The Portal showing `2.070 kWh` - which is a difference of 181 kWh! Which is almost 9% OFF...

Too bad that I was so foolish to rely on the data of the portal when I did my tax declarations (grid-export) - from now on I have to do this based on the real reading of the electricity-meter [good that I just have upgraded to Tibber-Pulse-IR with local reading]

Allow me to add finally, that I assume this is not really an issue with the mein-senec.de - it's more likely a issue with the way how this gird-import total value (kWh) is calculated based on the current grid import value - Even if now the local statistic data is gone - I would expect the same difference in the local data.

**Comment by Cyrelion** [comment]
Hi Marq24,

not only the mein-senec.de data aren't accurate, the local values (we had access in the past) from the senec battery system weren't too. Many of the consumption or produced values are "calculated" or not accurated measured. For example the grid power is most of the time different to the one I get from the calibrated energymeter of my provider. That's why I didn't use the grid export/import values from the senec system in my HA at all, I directly use the data of my electricity provider. Sad but true :-(

**Comment by bobthebuilder4711** [comment]
I wanted to merge the statistics data of the old sensors with the new sensors that rely on mein-senec.de.
Last entry of the old integration was at 3pm. First entry of the new sensors was at 5pm.
Between those two entries the total solar generation went down by almost 300kWh.

So whatever is going on, they seem to be guesstimating everything.

**Comment by Cyrelion** [comment]
The merge isn't that easy, it isn't just a remapping in the database to from the old id to the new one as the sum of the new values are the difference from the last value of that sensor. So to do a correct mapping you would need to get the last value of the old local entity and add this to the sum of the new one. After this is done you can change the ID of the old entry to the new ID.

**Comment by bobthebuilder4711** [comment]
I know. That's not really a problem. Simple UPDATE query to add that to the sum. That's why I was in the database in the first place. Because I thought that the data from senec wouldn't differ that wildly between the local total generation and what mein-senec.de reports.

But since they do differ by almost 300kWh this easy fix option isn't really viable anymore.

(And since I didn't have home assistant 4 years back when the senec battery was installed, I'm now thinking about using the statistics data on mein-senec.de, calculating kWh values from that and then using the websocket api to feed the recorder with that historical data. We will see how long it takes until I've got enough free time on my hands to write such a tool.)

**Comment by Cyrelion** [comment]
You could just take the last state value from before the 300 kWh and add this via sql to the newer sum values. This can be done via one sql query and should correct it.

**Comment by bobthebuilder4711** [comment]
No, I can't. The state of the entry at 3pm is 95713.12 and the state of the entry at 5pm is 95440.79. So 300kWh LESS in 2 hours. This has nothing to do with the sum.

But this isn't the right place to discuss this either. I just wanted to add that the local total values and the online total values differ from each other too.
